Branch-SDK/Branch-SDK/BNCSearchAdAttribution.m

```diff
@@ -0,0 +1,128 @@
+//
+//  BNCSearchAdAttribution.m
+//  Branch
+//
+//  Created by Edward Smith on 11/1/16.
+//
+
+
+#import "BNCSearchAdAttribution.h"
+#import <iAD/iAD.h>
+#import "BNCError.h"
+#import "BNCEncodingUtils.h"
+#import "BNCPreferenceHelper.h"
+
+
+/*
+    Flow:
+    
+    If the user has set 'checkAppleSearchAdInstall = YES' then Branch should instantiate this class
+    to check the value at start up.
+
+    Details:
+
+    1)  If it's been called before, the result is in preferences, and that value is returned.
+
+    2)  If hasn't been checked before, calls
+        
+        '[[ADClientClass sharedClient] requestAttributionDetailsWithBlock:]'
+
+        to get the result.  The result is saved in preferences.
+        
+- (void) setAppleSearchAdDetails:(NSDictionary*)details
+{
+    [self writeObjectToDefaults:@"appleSearchAdDetails" value:details];
+}
+
+- (NSDictionary*) appleSearchAdDetails
+{
+    return [self readObjectFromDefaults:@"appleSearchAdDetails"];
+}
+
+@property (assign, nonatomic) BOOL waitForAppleSearchAdDetails;
+
+*/
+
+
+@interface BNCSearchAdAttribution ()
+@end
+
+
+@implementation BNCSearchAdAttribution
+
++ (NSDictionary* _Nullable) lastAttribution
+{
+    return [BNCPreferenceHelper preferenceHelper].appleSearchAdDetails;
```

Comment:
I only want to make sure we exercise the lastAttributionWireFormatString function since it does a couple transformations